massa-models/src/block.rs

```diff
@@ -759,28 +835,54 @@ impl Deserializer<BlockHeader> for BlockHeaderDeserializer {
             parents[slot.thread as usize],
         ));
 
-        let (rest, endorsements) = context(
+        let (rest, endorsements): (&[u8], Vec<Wrapped<Endorsement, EndorsementId>>) = context(
             "Failed endorsements deserialization",
             length_count::<&[u8], Wrapped<Endorsement, EndorsementId>, u32, E, _, _>(
                 context("Failed length deserialization", |input| {
                     self.length_endorsements_deserializer.deserialize(input)
                 }),
                 context("Failed endorsement deserialization", |input| {
-                    endorsement_deserializer.deserialize_with(&self.endorsement_serializer, input)
+                    let (rest, endo) = endorsement_deserializer
+                        .deserialize_with(&self.endorsement_serializer, input)?;
+
+                    if endo.verify_signature().is_err() {
```

Comment:
After a discussion with @damip we remembered that we don't want to verify the signatures on the objects when in deserialization because we can receive objects multiple times sent by different peers and so we want to avoid calculating signatures for objects that we will drop few seconds after.

We prefer calculate the signatures after verifying uniqueness  [example with block header](https://github.com/massalabs/massa/blob/main/massa-protocol-worker/src/protocol_worker.rs#L839).

Reply:
so we don't want this in the block-header deserealization?

```rust
                    if endo.verify_signature().is_err() {
                        Err(nom::Err::Failure(ContextError::add_context(
                            rest,
                            "Endorsement signature invalid",
                            ParseError::from_error_kind(rest, nom::error::ErrorKind::Fail),
                        )))
                    } else {
                        Ok((rest, endo))
                    }
```

just return the the result of `deserialize_with`?

Reply:
We don't want to verify any signatures in the deserializers. We want to do it in others parts of the code to avoid calculate signatures of possible duplicate objects that we receive in the network

Reply:
Where should I check for signatures in that case?